src/cmd.js

```diff
@@ -1,117 +1,197 @@
+  // string: string | Array<string>
+  const cmdAliases = {
+    i: 'install',
+    sh: 'shell',
+    r: 'run',
+    'run-script': 'run',
+    't': {_: 'run', command: ['test']},
+    'test': {_: 'run', command: ['test']},
+    'start': {_: 'run', command: ['start']},
+    'stop': {_: 'run', command: ['stop']},
+    'build': {_: 'run', command: ['build']},
   }
 
-  if (argv.version) {
-    versionCmd = require('./version_cmd').default
-    return versionCmd()
+  if (argv.registry) {
+    config.registry = argv.registry
   }
 
-  const [subCommand] = argv._
-
-  switch (subCommand) {
-    case 'i':
-    case 'install':
-      installCmd = require('./install_cmd').default
-      installCmd(cwd, argv).subscribe()
-      break
-    case 'sh':
-    case 'shell':
-      shellCmd = require('./shell_cmd').default
-      shellCmd(cwd)
-      break
-    case 'r':
-    case 'run':
-    case 'run-script':
-      runCmd = require('./run_cmd').default
-      runCmd(cwd, argv)
-      break
-    case 't':
-    case 'test':
-    case 'start':
-    case 'build':
-    case 'stop':
-      runCmd = require('./run_cmd').default
-      const _argv = Object.create(argv)
-      _argv._ = ['run'].concat(argv._)
-      runCmd(cwd, _argv)
-      break
-    case 'ls':
-      lsCmd = require('./ls_cmd').default
-      lsCmd(cwd)
-      break
-    case 'ping':
-      pingCmd = require('./ping_cmd').default
-      pingCmd()
-      break
-    case 'conf':
-    case 'config':
-      configCmd = require('./config_cmd').default
-      configCmd()
-      break
-    case 'init':
-      initCmd = require('./init_cmd').default
-      initCmd(cwd, argv)
-      break
-    case 'link':
-      linkCmd = require('./link_cmd').default
-      linkCmd(cwd, argv)
-      break
-    case 'unlink':
-      unlinkCmd = require('./unlink_cmd').default
-      unlinkCmd(cwd, argv)
-      break
-    case 'cache':
-      cacheCmd = require('./cache_cmd').default
-      cacheCmd(cwd, argv)
-      break
-    case 'version':
-      versionCmd = require('./version_cmd').default
-      versionCmd()
-      break
-    default:
-      helpCmd = require('./help_cmd').default
-      helpCmd()
+  const inputCmd = argv._[0]
+  const alias = cmdAliases[inputCmd]
+  if (typeof alias === 'string') {
+    argv._ = [].concat(alias).concat(argv._.slice(1))
+  } else if (typeof alias === 'object' && alias._) {
+    for (const key in alias) {
+      argv[key] = alias[key]
+    }
   }
-})()
+  return argv
+}
+
+//
+// Run the commands!
+//
+
+const argv = resolveAliases(makeParser().argv)
+const cmd = argv._[0]
+
+switch (cmd) {
+  case 'install':
+    installCmd(cwd, argv, function (err) {
+      if (err) {
+        throw err
+      }
+    })
+    break
+  case 'help':
+    if (argv.command === 'install') {
+      // Turn `ied help install` into `ied install --help`
+      // Not easy to do with a cmdAlias so we transform it here.
+      // We can't tell yargs to re-parse, so we lie to it and create a new instance.
```

Comment:
Not that I know of. Yargs' method of showing help docs for individual commands is not expressed well. Outside of patching yargs ourselves, this seems to be the best way to go. It's very fast, though, and executed very seldom.